src/drivers/net/dwc_gmac/dwc_gmac.c

```diff
@@ -395,17 +401,17 @@ static uint32_t dwc_setup_rxdesc(struct dwc_priv *priv, int idx) {
 
 	assert(idx < RX_DESC_QUANTITY);
 
-	desc = &priv->rxdesc_ring_paddr[idx];
+	desc = &priv->rxdesc_ring[idx];
 
 	memset(desc, 0, sizeof(*desc));
 	data_mem_barrier();
 
-	desc->basic.des2 = (uint32_t)&rx_buffers[idx][0];
+	desc->basic.des2 = (uint32_t)dma_map_single(NULL, &rx_buffers[idx][0], ETH_FRAME_LEN, DMA_FROM_DEVICE);
 	desc->basic.des1 = ERDES1_SECOND_ADDRESS_CHAINED | ETH_FRAME_LEN;
 
 	idx++;
 	idx %= RX_DESC_QUANTITY;
-	desc->basic.des3 = (uint32_t)&priv->rxdesc_ring_paddr[idx];
+	desc->basic.des3 = (uint32_t)&((struct dma_extended_desc *)priv->rxdesc_ring_dma)[idx];
```

Comment:
Can this be replaced with something like `priv->rxdesc_ring_dma + idx * sizeof(struct dma_extended_desc)` to be more readable?